jobspec2/hcl_conversions.go

```diff
@@ -0,0 +1,281 @@
+package jobspec2
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/hashicorp/nomad/api"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+var hclDecoder *gohcl.Decoder
+
+func init() {
+	hclDecoder = newHCLDecoder()
+	hclDecoder.RegisterBlockDecoder(reflect.TypeOf(api.TaskGroup{}), decodeTaskGroup)
+}
+
+func newHCLDecoder() *gohcl.Decoder {
+	decoder := &gohcl.Decoder{}
```

Comment:
Obviously this compiles but just for my education where is this `Decoder` struct coming from? It's not in the [docs](https://pkg.go.dev/github.com/hashicorp/hcl/v2@v2.7.0/gohcl) and I can't find it in the `gohcl` source either!

Reply:
Yeah, I'm using a slightly modified hcl/v2 sha now - and will be working to getting the changes incorporated in hcl/v2 branch.  You can review those changes in https://github.com/hashicorp/hcl/compare/hcl2...hcl2-nomad-teaks .